jetty-jaspi in jetty-all uber aggregate artifact requires javax.security.auth.message.AuthException which cannot be included

**Jetty version**

Any version equal or greater than Jetty 9.4.23.

**Java version**

Any

**OS type/version**

Any

**Description**

Several projects (for instance Apache ActiveMQ) use jetty-all uber artifact (`org.eclipse.jetty.aggregate:jetty-all:uber`) doesn't include `org.eclipse.jetty.orbit:javax.security.auth.message`.
This missing artifact cause issue when using the `ConstraintSecurityHandler`:

```
Caused by: java.util.ServiceConfigurationError: org.eclipse.jetty.security.Authenticator$Factory: Provider org.eclipse.jetty.security.jaspi.JaspiAuthenticatorFactory could not be instantiated
  at java.util.ServiceLoader.fail(ServiceLoader.java:232)
  at java.util.ServiceLoader.access$100(ServiceLoader.java:185)
  at java.util.ServiceLoader$LazyIterator.nextService(ServiceLoader.java:384)
  at java.util.ServiceLoader$LazyIterator.next(ServiceLoader.java:404)
  at java.util.ServiceLoader$1.next(ServiceLoader.java:480)
  at org.eclipse.jetty.security.SecurityHandler.<clinit>(SecurityHandler.java:79)
  at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
  at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
  at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
  at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
  at org.eclipse.jetty.servlet.ServletContextHandler.newSecurityHandler(ServletContextHandler.java:318)
  at org.eclipse.jetty.servlet.ServletContextHandler.getSecurityHandler(ServletContextHandler.java:377)
  at org.eclipse.jetty.servlet.ServletContextHandler.relinkHandlers(ServletContextHandler.java:227)
  at org.eclipse.jetty.servlet.ServletContextHandler.<init>(ServletContextHandler.java:160)
  at org.eclipse.jetty.servlet.ServletContextHandler.<init>(ServletContextHandler.java:129)
  at org.apache.activemq.transport.ws.WSTransportServer.doStart(WSTransportServer.java:69)
  at org.apache.activemq.util.ServiceSupport.start(ServiceSupport.java:55)
  at org.apache.activemq.broker.TransportConnector.start(TransportConnector.java:250)
  at org.apache.activemq.broker.BrokerService.startTransportConnector(BrokerService.java:2761)
  at org.apache.activemq.broker.BrokerService.startAllConnectors(BrokerService.java:2654)
  at org.apache.activemq.broker.BrokerService.doStartBroker(BrokerService.java:777)
  at org.apache.activemq.broker.BrokerService.startBroker(BrokerService.java:739)
  at org.apache.activemq.broker.BrokerService.start(BrokerService.java:642)
  at org.apache.activemq.xbean.XBeanBrokerService.afterPropertiesSet(XBeanBrokerService.java:73)
  at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:498)
  at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeCustomInitMethod(AbstractAutowireCapableBeanFactory.java:1748)
  at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1685)
  at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1615)
  ... 27 more
Caused by: java.lang.NoClassDefFoundError: javax/security/auth/message/AuthException
  at java.lang.Class.getDeclaredConstructors0(Native Method)
  at java.lang.Class.privateGetDeclaredConstructors(Class.java:2671)
  at java.lang.Class.getConstructor0(Class.java:3075)
  at java.lang.Class.newInstance(Class.java:412)
  at java.util.ServiceLoader$LazyIterator.nextService(ServiceLoader.java:380)
  ... 55 more
Caused by: java.lang.ClassNotFoundException: javax.security.auth.message.AuthException
  at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
  at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
  at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
  ... 60 more
```

I see this exclude: https://github.com/eclipse/jetty.project/blob/f4fc78ac66a5811a68e264055114d879d74ae477/aggregates/jetty-all/pom.xml#L85

But it doesn't make sense to me, `jetty-all` should include `javax.security.auth.message.AuthException`. I'm creating a PR.

--- a/jetty-security/src/main/java/org/eclipse/jetty/security/SecurityHandler.java
+++ b/jetty-security/src/main/java/org/eclipse/jetty/security/SecurityHandler.java
@@ -24,8 +24,10 @@ import java.util.ArrayList;
 import java.util.Collection;
 import java.util.Enumeration;
 import java.util.HashMap;
+import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
+import java.util.ServiceConfigurationError;
 import java.util.ServiceLoader;
 import java.util.Set;
 import javax.servlet.ServletException;
@@ -76,9 +78,19 @@ public abstract class SecurityHandler extends HandlerWrapper implements Authenti
 
     static
     {
-        for (Authenticator.Factory factory : ServiceLoader.load(Authenticator.Factory.class))
+        Iterator<Authenticator.Factory> serviceLoaderIterator = ServiceLoader.load(Authenticator.Factory.class).iterator();
+        while (true)
         {
-            __knownAuthenticatorFactories.add(factory);
+            try
+            {
+                if (!serviceLoaderIterator.hasNext())
+                    break;
+                __knownAuthenticatorFactories.add(serviceLoaderIterator.next());
+            }
+            catch (ServiceConfigurationError error)
+            {
+                LOG.warn("Error while loading AuthenticatorFactory with ServiceLoader", error);
+            }
         }
 
         __knownAuthenticatorFactories.add(new DefaultAuthenticatorFactory());

